src/extension/app/components/plugin/plugin-action-bar.js

```diff
@@ -212,6 +120,13 @@ export class PluginActionBar extends MobxLitElement {
   }
 
   render() {
+    if (!this.#userPrefs) {
```

Comment:
Here is what I probably would have done in here. The call to this.#loadUserPrefs(); in render felt like it could be improved.

Just showing what I would change.

```javascript
  #userPrefs = null;

  /**
   * Are we ready to render?
   * @type {Boolean}
   */
  @property({ type: Boolean, attribute: false })
  accessor ready = false;

  async connectedCallback() {
    super.connectedCallback();

    const pluginSettings = await getConfig('sync', 'pluginPrefs') || {};
    this.#userPrefs = pluginSettings[appStore.getEnv()] || {};
    this.ready = true; //No need to call requestUpdate here since changing this variable will trigger a re-render.
  }
  
    render() {
      return this.ready ? html`
        <action-bar>
          ${this.renderPlugins()}
          ${this.renderSystemPlugins()}
        </action-bar>
    ` : '';
  }
```